### ISSUE TITLE
Bold syntax is broken when select text with a space in wysiwyg

### ISSUE THREAD

**Issue by sohee-lee7** [nhn/tui.editor#548]
## Version
v1.4.3

## Test Environment
Chrome

## Current Behavior

1. In Wysiwyg, select the test including space at the end.
![image](https://user-images.githubusercontent.com/42666492/59733712-caacae00-9289-11e9-8458-d0b3a842d057.png)

2. Apply bold using toolbar.
![image](https://user-images.githubusercontent.com/42666492/59733935-81a92980-928a-11e9-9d1b-65e5e3d86648.png)

3. Change mode to Markdown.
![image](https://user-images.githubusercontent.com/42666492/59733832-2b3beb00-928a-11e9-960c-5ecebb0c899b.png)

Bold syntax is broken in Markdown.

## Expected Behavior

Bold Syntax should not be broken.

**Comment by kcabtonmai** [comment]
This is happening for any formatted text that ends with space whenever converting to markdown.

**Comment by saeromCho** [comment]
Hello, I have the same issue. Both wysiwyg and markdown. 
I think it's related to issue #580. When will it be resolved?

**Comment by seonim-ryu** [comment]
@saeromCho 
The current milestone is to include this issue in version 1.4.7. I think distribution will be later this month.

**Comment by saeromCho** [comment]
> @saeromCho
> The current milestone is to include this issue in version 1.4.7. I think distribution will be later this month.

Thank you for your answer. 
If there's one more thing I need to tell you, I've tested a lot of situations, and the same thing happens in other situations besides the one above.

**Comment by seonim-ryu** [comment]
## Cause

If the html value entered in WYSIWYG is as below:

```html
<strong>foo </strong> bar
<b>foo </b> bar
<em>foo </em> bar

<i>foo </i> bar
<del>foo </del> bar

<s>foo </s> bar

foo <b>  bar</b> baz
```

When converted to markdown, the markdown is broken because it contains space characters between the emphasis(`**X**`, `*X*`, `~~X~~`)  syntax and the content.

```markdown
**foo ** bar
**foo ** bar

*foo * bar
*foo * bar

~~foo ~~ bar

foo **  bar** baz
```

## Solution

Move space characters between the syntax and the content out of the syntax. It should be handled at `to-mark`.

```markdown
**foo**  bar
**foo**  bar

*foo*  bar
*foo*  bar

~~foo~~  bar

foo   **bar** baz
```

**Comment by saeromCho** [comment]
Thank you for your kind reply even after the time has passed. 
As you said, I'll try to handle the `to-mark`. 
If there's another issue, I'll comment again. 
Thank you again for always being kind.

**Comment by seonim-ryu** [comment]
@saeromCho 
This issue was originally [fixed in the to-mark](https://github.com/nhn/to-mark/issues/36), I decided to move to editor logic. It will be included in v1.4.7 as scheduled. Please wait a bit longer.

**Comment by Typogram** [comment]
this doesn't seem to be fixed in the current build, in the editor view

https://ui.toast.com/tui-editor

It is very easy to select a trailing space. I think the best thing to do is to ignore that and treat it as if the seleectedText = selectedText.trim(), and then proceed to insert tags.

This is a code I have for a custom implementation, I don't know how to extend the toast ui to make this behavior happen?
```
    const startPos = textarea.selectionStart;
    const endPos = textarea.selectionEnd;
    const originalSelectedText = textarea.value.substring(startPos, endPos);

    // Calculate the difference between the original selected text and the trimmed version
    const leadingSpaces = originalSelectedText.length - originalSelectedText.trimStart().length;
    const trailingSpaces = originalSelectedText.length - originalSelectedText.trimEnd().length;

    // Update the selected text and positions based on the calculated differences
    const newStartPos = startPos + leadingSpaces;
    const newEndPos = endPos - trailingSpaces;
    const selectedText = textarea.value.substring(newStartPos, newEndPos);

    let newText = '';

    if (type === 'bold' || type === 'italic' || type === 'link' || type === 'image') {
      const tags = {
        bold: ['**', '**'],
        italic: ['_', '_'],
        link: ['[', ']'],
        image: ['![', ']'],
      };

      let url = '';

      if (type === 'link' || type === 'image') {
        url = prompt('Enter the URL:');
        if (!url) return;
        url = `(${url})`;
      }

      newText =
        tags[type][0] +
        selectedText +
        tags[type][1] +
        url;

      textarea.value =
        textarea.value.substring(0, newStartPos) +
        newText +
        textarea.value.substring(newEndPos);

      textarea.selectionEnd = newEndPos + tags[type][0].length + tags[type][1].length + url.length;
    }
  ```
This ignore space selectedText behavior can be seen in this Github Issue editor, I highly recommend having this behavior.
